tbx/core/tests/test_colour_theme.py

```diff
@@ -0,0 +1,208 @@
+
+        work = WorkPageFactory(parent=self.workindex)
+        self.assertEqual(work.theme_class, ColourTheme.NONE)
+
+        std_page = StandardPageFactory(parent=work)
+        self.assertEqual(std_page.theme_class, ColourTheme.NONE)
+
+        # -------------------------------------------
+        # Now we check the templates for css classes
+        # -------------------------------------------
+
+        for page in [self.home, self.blogindex, self.workindex, work, blog, std_page]:
+            class_suffix = slugify(page.get_verbose_name())
+            response = self.client.get(page.url)
+            soup = BeautifulSoup(response.content, "html.parser")
+
+            # Get the list of classes on the html tag
+            html_classes = soup.find("html").get("class", [])
+
+            # Check if the specified class is the only class present
+            self.assertEqual(len(html_classes), 1)
+            self.assertEqual(html_classes[0], f"template-{class_suffix}")
+
+    def test_theme_on_homepage(self):
+        self.home.theme = ColourTheme.CORAL
+        self.home.save()
+        self.home.refresh_from_db()
+
+        self.assertEqual(self.home.theme_class, ColourTheme.CORAL)
+        self.assertEqual(self.blogindex.theme_class, ColourTheme.CORAL)
+        self.assertEqual(self.workindex.theme_class, ColourTheme.CORAL)
+
+        blog = BlogPageFactory(parent=self.blogindex)
+        self.assertEqual(blog.theme_class, ColourTheme.CORAL)
+
+        work = WorkPageFactory(parent=self.workindex)
+        self.assertEqual(work.theme_class, ColourTheme.CORAL)
+
+        std_page = StandardPageFactory(parent=work)
+        self.assertEqual(std_page.theme_class, ColourTheme.CORAL)
+
+        # -------------------------------------------
+        # Now we check the templates for css classes
+        # -------------------------------------------
+
+        for page in [self.home, self.blogindex, self.workindex, work, blog, std_page]:
+            class_suffix = slugify(page.get_verbose_name())
+            response = self.client.get(page.url)
+            soup = BeautifulSoup(response.content, "html.parser")
+
+            # Get the list of classes on the html tag
+            html_classes = soup.find("html").get("class", [])
+
+            # Check that we have the correct classes applied
+            self.assertEqual(len(html_classes), 2)
+            self.assertEqual(html_classes[0], f"template-{class_suffix}")
+            self.assertEqual(html_classes[1], ColourTheme.CORAL)
+
+    def test_theme_on_an_indexpage(self):
+        self.blogindex.theme = ColourTheme.LAGOON
+        self.blogindex.save()
+        self.blogindex.refresh_from_db()
+
+        self.assertEqual(self.home.theme_class, ColourTheme.NONE)
+        self.assertEqual(self.blogindex.theme_class, ColourTheme.LAGOON)
+
+        # child of blogindex
+        blog = BlogPageFactory(parent=self.blogindex)
+        self.assertEqual(blog.theme_class, ColourTheme.LAGOON)
```

Comment:
**Suggestion**: It may be helpful to add a check of the field directly, to ensure that the inheritance is in use here.

```suggestion
        # child of blogindex
        blog = BlogPageFactory(parent=self.blogindex)
        self.assertEqual(blog.theme, ColourTheme.NONE)
        self.assertEqual(blog.theme_class, ColourTheme.LAGOON)
```

Reply:
Thanks, updated in b527c3b